### ISSUE TITLE
Post processing instance segmentation taking a lot of time

### ISSUE THREAD

**Issue by lwillems191** [NickSwardh/YoloDotNet#14]
Hello,

When using instance segmentation the post processing is taking quite a lot of time and I was wondering if there might be way to optimize it. I found which line is taking the most time, but have not found a way to optimize it. 
Maybe somebody else has a good idea.

`var value = Enumerable.Range(0, output.Channels).Sum(i => tensor1[0, i, y, x] * maskWeights[i]);`

**Comment by aloksharma1** [comment]
can you try this? (untested but changing to a for loop would surely improve it)

```
float value = 0;
for (int i = 0; i < output.Channels; i++)
{
    value += tensor1[0, i, y, x] * maskWeights[i];
}
```

**Comment by aloksharma1** [comment]
> Currently working on improving the overall performance in this branch https://github.com/NickSwardh/YoloDotNet/tree/performance where I've replaced the usage a legacy ONNX-class to use OrtValue Api instead for improved performance along with a few other tweaks here and there.

is this branch prod ready (on nuget)?

**Comment by lwillems191** [comment]
> Currently working on improving the overall performance in this branch https://github.com/NickSwardh/YoloDotNet/tree/performance where I've replaced the usage a legacy ONNX-class to use OrtValue Api instead for improved performance along with a few other tweaks here and there.

Yeah this branch already gives a great improvement. Thanks for the work you put into it.

**Comment by NickSwardh** [comment]
Awesome! Thank you for letting me know :)

**Comment by NickSwardh** [comment]
> > Currently working on improving the overall performance in this branch https://github.com/NickSwardh/YoloDotNet/tree/performance where I've replaced the usage a legacy ONNX-class to use OrtValue Api instead for improved performance along with a few other tweaks here and there.
> 
> is this branch prod ready (on nuget)?

No, not yet, It's a work in progress. I'm still turning the nuts and bolts to see if I can squeeze some more speed out of this thing ;)

**Comment by lwillems191** [comment]
I did some more testing. The new improvements make the code already a lot faster, but from my testing it seems it might be better to not use Parellel.For loops. They seem to be a lot less consistent then a normal for loop. Also the speed improvement does not seem that much. Hopefully you will take this into consideration.
